features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesView.kt

```diff
@@ -472,10 +473,17 @@ private fun MessagesViewComposerBottomSheetContents(
                 // Do not show the identity change if user is composing a Rich message or is seeing suggestion(s).
                 if (state.composerState.suggestions.isEmpty() &&
                     state.composerState.textEditorState is TextEditorState.Markdown) {
-                    IdentityChangeStateView(
-                        state = state.identityChangeState,
-                        onLinkClick = onLinkClick,
-                    )
+                    if (state.identityChangeState.roomMemberIdentityStateChanges.isNotEmpty()) {
+                        IdentityChangeStateView(
+                            state = state.identityChangeState,
+                            onLinkClick = onLinkClick,
+                        )
+                    } else {
+                        HistoryVisibleStateView(
+                            state = state.historyVisibleState,
+                            onLinkClick = onLinkClick,
+                        )
+                    }
```

Comment:
Thanks for the fix. It's OK like that, even if it can make reader feel that `HistoryVisibleStateView` has no condition to be displayed (I know the method can return early if nothing has to be displayed)